woocommerce/woocommerce-ios
# Product Sharing: UI for generating share message with AI

Part of #9867 


## Description

This PR adds UI for the AI-generated sharing message for products feature following the latest update in pe5sF9-1AH-p2#comment-2256. Changes include:
- `ProductSharingMessageGenerationView` and view model: the UI for the new sharing sheet. 
- `FirstProductCreatedHostingController`, `ProductFormViewController`, `ProductsViewController`: persist `shareProductCoordinator` locally so that the coordinator is not deallocated after the containing methods are done executing.
- `ShareProductCoordinator`: Added a bottom sheet containing the new share sheet. 

## Testing instructions

Prerequisite: Make sure that you have access to a live WPCom site.
- On the Products tab, swipe to share a product or tap to share from its detail screen.
- Notice that a bottom sheet is displayed for AI-generated messages.
- Tap the button for generating message, notice that the loading state is displayed.
- When a message is generated, it should be displayed in a text editor. You should be able to change the content of the message.
- Tap Regenerate, a new message should be generated. Note: if you regenerate within 30 seconds of the last request, you'll get cached result from the last request.
- Tap Share, notice that the sheet is dismissed, and a native share sheet is displayed, and you can share both the product link and the generated message.

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/5533851/117e5d9a-d58c-4068-8592-1df79969b58b





---
